pwsh_module/pwsh_bolt.tests.ps1.erb

```diff
@@ -1,84 +1,87 @@
-$here = Split-Path -Parent $MyInvocation.MyCommand.Path
-$sut = (Split-Path -Leaf $MyInvocation.MyCommand.Path) -replace '\.tests\.', '.'
-. "$here\$sut"
-
-Describe "<%= @pwsh_command[:cmdlet] -%>" {
-
-  $commandName = '<%= @pwsh_command[:cmdlet] -%>'
-  $command = Get-Command -Name $commandName
+. $PSCommandPath.Replace('.tests.ps1', '.ps1')
+$commandName = '<%= @pwsh_command[:cmdlet] -%>'
+$command = Get-Command $commandName
+$help = Get-Help -Name $commandName
+
+$common = 'Debug', 'ErrorAction', 'ErrorVariable', 'InformationAction', 'InformationVariable', 'OutBuffer',
+'OutVariable', 'PipelineVariable', 'Verbose', 'WarningAction', 'WarningVariable', 'Confirm', 'Whatif'
+
+## Without the filter, WhatIf and Confirm parameters are still flagged in "finds help parameter in code" test
+$helpParameters = $help.Parameters.Parameter |
+  Where-Object { $_.Name -notin $common } |
+  Sort-Object -Property Name -Unique
+
+$commandParameters = $command.ParameterSets.Parameters |
+  Sort-Object -Property Name -Unique |
+  Where-Object { $_.Name -notin $common }
+
+$helpCases = New-Object System.Collections.Generic.List[PSObject]
+
+foreach ($parameter in $helpParameters) {
+  $currentCase = @{
+    commandName            = $commandName
+    parameterName          = $parameter.Name
+    parameterHelp          = $help.parameters.parameter | Where-Object Name -EQ $parameter.Name
+    paramTypeName          = $parameter.Type.Name
+    commandSaysisMandatory = $commandParameters | Where-Object name -eq $parameter.name | Select-Object -ExpandProperty IsMandatory
+  }
+  $helpCases.Add($currentCase) | Out-Null
+}
 
-  It "has correct parameters" {
+Describe '<%= @pwsh_command[:cmdlet] -%>' {
+  Context 'parameters' {
+    # Pester 5 doesn't allow outside variables to leak into the `It` block scope
+    # anymore. Until the Pester project has a supported workflow for this the
+    # workaround is to inject the data the block need via a single test case.
+    It 'has correct parameters' -TestCases @{command = $command} {
```

Comment:
Strictly speaking, if you used a BeforeAll block to create the `$command` and `$help` parameters, they would be in scope for the `It`  blocks that need them. But, creating the `$helpCases` variable in that same block would mean it's not created when the `foreach()` block below needs to create the tests. Since the `$helpCases` variable can't be in a `Before` block, and it depends on the `$command` and `$help` variables, then they can't be in a before block either, and now we're stuck using this annoying trick to get those two into the it block scope.